dhcpv4/client.go

```diff
@@ -165,14 +207,110 @@ func (c *Client) Exchange(ifname string, discover *DHCPv4, modifiers ...Modifier
 	conversation = append(conversation, request)
 
 	// Ack
-	ack, err := BroadcastSendReceive(sfd, rfd, request, c.ReadTimeout, c.WriteTimeout, MessageTypeAck)
+	ack, err := c.SendReceive(sfd, rfd, request, MessageTypeAck)
 	if err != nil {
 		return conversation, err
 	}
 	conversation = append(conversation, ack)
 	return conversation, nil
 }
 
+// SendReceive sends a packet (with some write timeout) and waits for a
+// response up to some read timeout value. If the message type is not
+// MessageTypeNone, it will wait for a specific message type
+func (c *Client) SendReceive(sendFd, recvFd int, packet *DHCPv4, messageType MessageType) (*DHCPv4, error) {
```

Comment:
Maybe it is a good idea to unexport this as in the v6 client so that only Exchange is used.

Reply:
since it's part of the public interface, before unexporting this should have a deprecation warning for a while. We can open an issue to prune deprecated public functions

Reply:
Will do.